src/Control/Effect/NonDet.hs

```diff
@@ -78,6 +79,10 @@ instance MonadTrans NonDetC where
   lift m = NonDetC (\ cons nil -> m >>= flip cons nil)
   {-# INLINE lift #-}
 
+instance MonadBase b m => MonadBase b (NonDetC m) where
```

Comment:
It does not appear that there can be a valid `MonadTransControl` instance for `NonDetC`, or indeed any other carrier that carries a reference to the base monad `m` in its state, as this makes it impossible to implement the `StM` type function. In addition, `NonDetC` results in an arbitrary `Alternative`, so `StT` is not possible (at least not without a `SomeAlternative` existential).